data-processor/src/test/groovy/io/micronaut/data/processor/sql/BuildQuerySpec.groovy

```diff
@@ -150,6 +151,47 @@ interface MealRepository extends CrudRepository<Meal, Long> {
 
     }
 
+    void "test join query on collection via association entity"() {
+        given:
+        def repository = buildRepository('test.BookRepository', """
+import io.micronaut.context.annotation.Executable;
+import io.micronaut.data.annotation.Join;
+import io.micronaut.data.jdbc.annotation.JdbcRepository;
+import io.micronaut.data.model.query.builder.sql.Dialect;
+import io.micronaut.data.repository.GenericRepository;
+import io.micronaut.data.tck.entities.Author;
+import io.micronaut.data.tck.entities.Book;
+
+import java.util.Optional;
+
+@JdbcRepository(dialect = Dialect.H2)
+interface BookRepository extends GenericRepository<Book, Long> {
+
+    @Join(value = "author.books", type = Join.Type.LEFT_FETCH)
+    public abstract Author findAuthorById(@Id Long id);
+
+    public abstract List<Book> findAuthorBooksById(@Id Long id);
+}
+""")
+
+
+        def authorMethod = repository.getRequiredMethod("findAuthorById", Long)
+        def authorQuery = getQuery(authorMethod)
+        def authorJoins = getJoins(authorMethod)
+
+        def booksMethod = repository.getRequiredMethod("findAuthorBooksById", Long)
+        def booksQuery = getQuery(booksMethod)
+        def booksJoins = getJoins(booksMethod)
+
+        expect:
+        authorQuery == "SELECT book_author_.`id`,book_author_.`name`,book_author_.`nick_name`,book_author_books_.`id` AS books_id,book_author_books_.`author_id` AS books_author_id,book_author_books_.`genre_id` AS books_genre_id,book_author_books_.`title` AS books_title,book_author_books_.`total_pages` AS books_total_pages,book_author_books_.`publisher_id` AS books_publisher_id,book_author_books_.`last_updated` AS books_last_updated FROM `book` book_ LEFT JOIN `author` book_author_ ON book_.`author_id`=book_author_.`id` LEFT JOIN `book` book_author_books_ ON book_author_.`id`=book_author_books_.`author_id` WHERE (book_.`id` = ?)"
+        authorJoins.size() == 1
+        authorJoins.keySet() == ["books"] as Set
+
+        booksQuery == "SELECT book_author_books_.`id`,book_author_books_.`author_id`,book_author_books_.`genre_id`,book_author_books_.`title`,book_author_books_.`total_pages`,book_author_books_.`publisher_id`,book_author_books_.`last_updated` FROM `book` book_ INNER JOIN `author` book_author_ ON book_.`author_id`=book_author_.`id` INNER JOIN `book` book_author_books_ ON book_author_.`id`=book_author_books_.`author_id` WHERE (book_.`id` = ?)"
```

Comment:
Why are there joins and the method is not annotated with any join and `booksJoins.size() == 0`

Reply:
It gets created through QueryCriteriaMethodMatch
```
query.multiselect(selectionList);
```
where selectionList is the association Author->books and then this line `selection.accept(joiner);` 
```
if (selection instanceof SelectionVisitable) {
            SelectionVisitable selection = (SelectionVisitable) this.selection;
            selection.accept(new QueryModelSelectionVisitor(qm, distinct));
            selection.accept(joiner);
            SelectionVisitable entityRoot = (SelectionVisitable) this.entityRoot;
            entityRoot.accept(joiner);
        }
```
in AbstractPersistentEntityCriteriaQuery.getQueryModel populates join. And finally gets processed in AbstractSqlLikeBuilder buildSelect method via projections. 
```
String propertyName = pp.getPropertyName();
                        PersistentPropertyPath propertyPath = entity.getPropertyPath(propertyName);
```
propertyName here is author.books and it here is is being added as join since QueryModel has got join paths 
```
Collection<JoinPath> joinPaths = queryState.getQueryModel().getJoinPaths();
```